packages/unmock-core/src/matcher.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  IMock,
+  IMockRequest,
+  ISerializedRequest,
+  ISerializedResponse,
+} from "./interfaces";
+
+type RequestComparator = (
+  intercepted: ISerializedRequest,
+  mock: IMockRequest,
+) => boolean;
+
+function propertyMatches({
+  property,
+  interceptedRequest,
+  mockRequest,
+}: {
+  property: keyof (ISerializedRequest & IMockRequest);
+  interceptedRequest: ISerializedRequest;
+  mockRequest: IMockRequest;
+}): boolean {
+  const mockRequestProperty = mockRequest[property];
+  const interceptedRequestProperty = interceptedRequest[property];
+
+  if (mockRequestProperty === undefined) {
+    return true;
```

Comment:
Why return true on undefined property? I'd expect false by nature (no match if property doesn't exist?)

Reply:
Hmm, to me it seemed most natural that if e.g. protocol is not defined, then it matches both http and https? Otherwise, it would be a bit cumbersome to define matching for both http and https. And by that token, I thought it would be simplest if the same "match for undefined" was true for every property.

Reply:
I can see how that's true for protocol, but what about `body` or `headers`?

I'm fine  with "match for undefined", but I think it needs to be explicitly mentioned somewhere then.